lib/agent0/examples/interactive_econ.py

```diff
@@ -0,0 +1,453 @@
+larry.add_liquidity(base=FixedPoint(exp.amount_of_liquidity))  # 10 million
+rob = interactive_hyperdrive.init_agent(base=FixedPoint(exp.amount_of_liquidity), name="rob")
+print(f"spot price = {interactive_hyperdrive.hyperdrive_interface.calc_spot_price()}")
+
+# %%
+# do some trades
+Max = NamedTuple("Max", [("base", FixedPoint), ("bonds", FixedPoint)])
+GetMax = NamedTuple("GetMax", [("long", Max), ("short", Max)])
+
+start_time = time.time()
+
+
+def get_max(
+    _interactive_hyperdrive: InteractiveHyperdrive, _share_price: FixedPoint, _current_base: FixedPoint
+) -> GetMax:
+    """Get max trade sizes.
+
+    Returns
+    -------
+    GetMax
+        A NamedTuple containing the max long in base, max long in bonds, max short in bonds, and max short in base.
+    """
+    _max_long_base = _interactive_hyperdrive.hyperdrive_interface.calc_max_long(budget=_current_base)
+    _max_long_shares = _interactive_hyperdrive.hyperdrive_interface.calc_shares_out_given_bonds_in_down(_max_long_base)
+    _max_long_bonds = _max_long_shares * _share_price
+    _max_short_bonds = FixedPoint(0)
+    try:  # sourcery skip: do-not-use-bare-except
+        _max_short_bonds = _interactive_hyperdrive.hyperdrive_interface.calc_max_short(budget=_current_base)
+    except:  # pylint: disable=bare-except
+        pass
+    max_short_shares = _interactive_hyperdrive.hyperdrive_interface.calc_shares_out_given_bonds_in_down(
+        _max_short_bonds
+    )
+    _max_short_base = max_short_shares * _share_price
+    return GetMax(
+        Max(_max_long_base, _max_long_bonds),
+        Max(_max_short_base, _max_short_bonds),
+    )
+
+
+# sourcery skip: avoid-builtin-shadow, do-not-use-bare-except, invert-any-all,
+# remove-unnecessary-else, swap-if-else-branches
+for day in range(exp.term_days):
+    amount_to_trade_base = FixedPoint(exp.amount_of_liquidity * exp.daily_volume_percentage_of_liquidity)
+    while amount_to_trade_base > MINIMUM_TRANSACTION_AMOUNT:
+        randnum = np.random.randint(0, 2)
+        spot_price = interactive_hyperdrive.hyperdrive_interface.calc_spot_price()
+        share_price = interactive_hyperdrive.hyperdrive_interface.current_pool_state.pool_info.share_price
+        max = None
+        wallet = rob.wallet
+        event = None
+        if randnum == 0:  # go long
+            if len(wallet.shorts) > 0:  # check if we have shorts, and close them if we do
+                for maturity_time, short in wallet.shorts.copy().items():
+                    max = get_max(interactive_hyperdrive, share_price, rob.wallet.balance.amount)
+                    amount_to_trade_bonds = (
+                        interactive_hyperdrive.hyperdrive_interface.calc_bonds_out_given_shares_in_down(
+                            amount_to_trade_base / share_price
+                        )
+                    )
+                    trade_size_bonds = min(amount_to_trade_bonds, short.balance, max.long.bonds)
```

Comment:
what is this accounting for? it seems like it is redundant with `get_max`, since the underlying `calc_max_long` already computes a min over the pool max & the wallet max.

Reply:
```
amount_to_trade_bonds = amount needed to trade to hit the volume target on this day
short.balance = size of my short that I'm considering decreasing
max.long.bonds = the output of calc_max_long which also tells me how much I can reduce my short (both add bonds into the pool)
```